std/jvm/src/main/scala/cats/effect/std/SecureRandomCompanionPlatform.scala

```diff
@@ -14,8 +14,120 @@
  * limitations under the License.
  */
 
-package cats.effect.std
+package cats
+package effect
+package std
+
+import cats._
+import cats.effect.kernel._
+import cats.effect.std.Random.ScalaRandom
+
+import scala.util.{Random => SRandom, Try}
+
+import java.util.concurrent.atomic.AtomicInteger
 
 private[std] trait SecureRandomCompanionPlatform {
   private[std] type JavaSecureRandom = java.security.SecureRandom
+  private[std] def getInstance: JavaSecureRandom =
+    java.security.SecureRandom.getInstance("NativePRNGNonBlocking")
+
+  private def javaUtilRandom[F[_]: Sync](random: JavaSecureRandom): SecureRandom[F] =
+    new ScalaRandom[F](Applicative[F].pure(random)) with SecureRandom[F] {}
+
+  /**
+   * Creates a blocking Random instance.
+   *
+   * @param random
+   *   a potentially blocking instance of java.util.Random
+   */
+  def javaUtilRandomBlocking[F[_]: Sync](random: JavaSecureRandom): SecureRandom[F] =
```

Comment:
This can be private like `javaUtilRandom` above.

```suggestion
  private def javaUtilRandomBlocking[F[_]: Sync](random: JavaSecureRandom): SecureRandom[F] =
```